hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieDefaultTimeline.java

```diff
@@ -228,9 +228,10 @@ public HoodieDefaultTimeline findInstantsInRangeByCompletionTime(String startTs,
   @Override
   public HoodieDefaultTimeline findInstantsModifiedAfterByCompletionTime(String instantTime) {
     return new HoodieDefaultTimeline(instants.stream()
-        .filter(s -> s.getCompletionTime() != null
-            && HoodieTimeline.compareTimestamps(s.getCompletionTime(), GREATER_THAN, instantTime)
-            && !s.getTimestamp().equals(instantTime)), details);
+        // either pending or completionTime greater than instantTime
+        .filter(s -> (s.getCompletionTime() == null && compareTimestamps(s.getTimestamp(), GREATER_THAN, instantTime))
+            || (compareTimestamps(s.getCompletionTime(), GREATER_THAN, instantTime) && !s.getTimestamp().equals(instantTime))),
```

Comment:
The `s.getCompletionTime() != null` check is missing?

Reply:
my bad.. i meant to keep it.. added now. thanks for pointing out.